### ISSUE TITLE
Keyboard DataGrid navigation broken when `FilterMode=FilterMode.Simple`

### ISSUE THREAD

**Issue by JustASquid** [radzenhq/radzen-blazor#1761]
**Describe the bug**
When `FilterMode` is `Simple`, keyboard navigation doesn't work properly.

**To Reproduce**
1. Go to https://blazor.radzen.com/datagrid
2. Replace FilterMode with `FilterMode.Simple`
3. Navigate with up and down keyboard keys and enter

**Expected behavior**
Navigation works the same as other filter modes

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Firefox
 - Version N/A
